vagrant/setup.sh

```diff
@@ -1,22 +1,39 @@
-#!/bin/bash
-
-# Commands required to setup working docker environment, link containers etc.
-sudo apt-get update
-sudo apt-get install -qy ruby1.9.1-dev build-essential sqlite3 libsqlite3-dev
-sudo gem install mailcatcher
-
-# Stop and remove any existing containers
-docker stop $(docker ps -a -q)
-docker rm $(docker ps -a -q)
-
-# Removed if already prepared
-rm -rf ./tmp/container_prepared
-
-# Run and link the containers
-docker run -d -p 5432:5432 -v /vagrant:/vagrant -e "POSTGRES_PASSWORD=postgres" --name postgres postgres:9.4.1
-docker run -d -p 3000:3000 -v /vagrant:/var/www --privileged --link postgres:db \
-        -e "RAILS_ENV=development" \
-        -e "DATABASE_URL=postgres://postgres:postgres@db:5432/ecommerce" \
-        --name rails ylosix/ylosix ./vagrant/start_container.sh
-
-docker exec rails ./vagrant/prepare_container.sh
+#!/bin/bash
+
+# Stop and remove any existing containers
+docker stop $(docker ps -a -q)
+docker rm $(docker ps -a -q)
+
+# Run postgres container
+docker run -d -p 5432:5432 -v /vagrant:/vagrant -e "POSTGRES_PASSWORD=postgres" --name postgres postgres:9.4.1
+
+# install required packages
+sudo apt-get update
+sudo apt-get install -qy git
+sudo apt-get -qy install nodejs
+sudo apt-get install -qy imagemagick
+#sudo gem install mailcatcher
+sudo apt-get install -qy libpq-dev   # dependency for pg gem
+sudo apt-get install -qy postgresql-client
+
+# To start using RVM you need to run `source /home/vagrant/.rvm/scripts/rvm`
+# in all your open shell windows, in rare cases you need to reopen all shell windows.
+source /home/vagrant/.rvm/scripts/rvm
+rm -rf /usr/local/bundle/config
+
+gem update --system
+rvm gemset use global
+gem update
+# disable gem documentation install to speed up gem installation
+echo "gem: --no-rdoc --no-ri" >> ~/.gemrc
+gem install bundler
+
+# Set the working environment
+source ~/.profile && [ -z "$DATABASE_URL" ] && \
+    echo "export DATABASE_URL=postgres://postgres:postgres@127.0.0.1:5432/ecommerce" >> ~/.profile
+
+# generate database
+cd /vagrant
+bundle install

```

Comment:
Rewrite these line as:
cd /vagrant; bundle install
cd /vagrant; bundle exec rake db:create db:schema:load db:gen_demo

Same in the start.sh